facade/src/main/java/org/jboss/pnc/facade/providers/DeliverableAnalyzerReportProviderImpl.java

```diff
@@ -0,0 +1,108 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.facade.providers;
+
+import com.google.common.collect.ObjectArrays;
+import org.jboss.pnc.dto.response.AnalyzedArtifact;
+import org.jboss.pnc.dto.response.Page;
+import org.jboss.pnc.facade.providers.api.DeliverableAnalyzerReportProvider;
+import org.jboss.pnc.mapper.api.ArtifactMapper;
+import org.jboss.pnc.mapper.api.DeliverableAnalyzerReportMapper;
+import org.jboss.pnc.model.Base32LongID;
+import org.jboss.pnc.model.DeliverableAnalyzerReport;
+import org.jboss.pnc.model.DeliverableArtifact;
+import org.jboss.pnc.spi.datastore.predicates.DeliverableArtifactPredicates;
+import org.jboss.pnc.spi.datastore.repositories.DeliverableAnalyzerReportRepository;
+import org.jboss.pnc.spi.datastore.repositories.DeliverableArtifactRepository;
+import org.jboss.pnc.spi.datastore.repositories.api.PageInfo;
+import org.jboss.pnc.spi.datastore.repositories.api.Predicate;
+import org.jboss.pnc.spi.datastore.repositories.api.SortInfo;
+
+import javax.annotation.security.PermitAll;
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@PermitAll
+@Stateless
+public class DeliverableAnalyzerReportProviderImpl extends
+        AbstractProvider<Base32LongID, DeliverableAnalyzerReport, org.jboss.pnc.dto.DeliverableAnalyzerReport, org.jboss.pnc.dto.DeliverableAnalyzerReport>
+        implements DeliverableAnalyzerReportProvider {
+
+    private DeliverableAnalyzerReportRepository deliverableAnalyzerReportRepository;
+
+    private DeliverableArtifactRepository deliverableArtifactRepository;
+
+    private DeliverableAnalyzerReportMapper deliverableAnalyzerReportMapper;
+
+    private ArtifactMapper artifactMapper;
+
+    @Inject
+    public DeliverableAnalyzerReportProviderImpl(
+            DeliverableAnalyzerReportRepository repository,
+            DeliverableArtifactRepository deliverableArtifactRepository,
+            DeliverableAnalyzerReportMapper mapper,
+            ArtifactMapper artifactMapper) {
+        super(repository, mapper, DeliverableAnalyzerReport.class);
+
+        this.deliverableAnalyzerReportRepository = repository;
+        this.deliverableArtifactRepository = deliverableArtifactRepository;
+        this.deliverableAnalyzerReportMapper = mapper;
+        this.artifactMapper = artifactMapper;
+    }
+
+    @Override
+    public Page<AnalyzedArtifact> getAnalyzedArtifacts(
+            int pageIndex,
+            int pageSize,
+            String query,
+            String sort,
+            String id) {
+        Base32LongID entityId = mapper.getIdMapper().toEntity(id);
+        Predicate<DeliverableArtifact> rsqlPredicate = rsqlPredicateProducer
+                .getCriteriaPredicate(DeliverableArtifact.class, query);
```

Comment:
Doesn't this require a `DeliverableArtifactRSQLMapper`?

Reply:
Nice catch Honza!!
The `DeliverableArtifactRSQLMapper` was introduced, see [1df735d](https://github.com/project-ncl/pnc/pull/3816/commits/1df735d57b28f9cb6b2c876743f500b558ed020f).
Please, do think about the fact, that the primary key of `DeliverableArtifact` is the composite key `DeliverableArtifactPK` and whether this composite key cannot potentially screw some things in your RSQL abstraction machinery up.